src/webhook_launcher/app/tasks.py

```diff
@@ -87,20 +93,28 @@ def trigger_build(mapobj, user, lsr=None, tag=None, force=False):
                 delayed = True
                 break
 
+    # Updat LastSeenRevision.tag if this is going to be built
+    if build and not skipped and tag:
+        lsr.tag = tag
+
     # handle_build actually launches a build process
     msg = handle_build(mapobj, user, lsr, force, skipped, delayed, qp)
```

Comment:
Why do we call handle_build if build is False? Either the naming is misleading or we shouldn't? The handle_build code then does various things..